Add definition of ISA circuits (#1042)
Qiskit/documentation: docs/transpile/index.mdx

Transpilation is the process of rewriting a given input circuit to match the topology of a specific quantum device, and optimize the circuit instructions for execution on noisy quantum systems.  This documentation covers the tooling and workflows for local transpilation available to all Qiskit users, as well as for the cloud-based [Qiskit transpiler service](qiskit-transpiler-service) available to Premium Plan users.  If you're using primitives and are only interested in the default transpilation options provided by the Qiskit Runtime service, read the [Configure runtime compilation for Qiskit Runtime](../run/configure-runtime-compilation) topic.

> A central component of the Qiskit SDK, the transpiler is designed for modularity and extensibility. Its central goal is to write new circuit transformations (known as transpiler **passes**), and combine them with other existing passes, greatly reducing the depth and complexity of quantum circuits.  Which passes are chained together and in which order has a major effect on the final outcome. This pipeline is determined by the [`PassManager`](/api/qiskit/qiskit.transpiler.PassManager) and [`StagedPassManager`](/api/qiskit/qiskit.transpiler.StagedPassManager) objects.  The `StagedPassManager` orchestrates the execution of one or more `PassManagers` and determines the order in which they are executed, while the `PassManager` object is merely a collection of one or more passes.  Think of the `StagedPassManager` as the conductor in an orchestra, the `PassManagers` as the different instrument sections, and the `Pass` objects as the individual musicians.  In this way, you can compose hardware-efficient quantum circuits that let you execute utility-scale work while keeping noise manageable.

Reviewer: ```suggestion
A central component of the Qiskit SDK, the transpiler is designed for modularity and extensibility. Its main goal is to write new circuit transformations (known as transpiler **passes**), and combine them with other existing passes, greatly reducing the depth and complexity of quantum circuits.  Which passes are chained together and in which order has a major effect on the final outcome. This pipeline is determined by the [`PassManager`](/api/qiskit/qiskit.transpiler.PassManager) and [`StagedPassManager`](/api/qiskit/qiskit.transpiler.StagedPassManager) objects.  The `StagedPassManager` orchestrates the execution of one or more `PassManagers` and determines the order in which they are executed, while the `PassManager` object is merely a collection of one or more passes.  Think of the `StagedPassManager` as the conductor in an orchestra, the `PassManagers` as the different instrument sections, and the `Pass` objects as the individual musicians.  In this way, you can compose hardware-efficient quantum circuits that let you execute utility-scale work while keeping noise manageable.
```

Reviewer: two "central"s in consecutive sentences :)

Find more information about the pass manager stages in the [Transpiler stages](transpiler-stages) topic.

> ### Instruction Set Architecture
> In addition to reducing the depth and complexity of quantum circuits, the transpiler is designed to transform the instructions contained in a given `QuantumCircuit` to obey the **Instruction Set Architecture** (**ISA**) of a particular backend.  Circuits obeying the ISA consist only of gates that can be run on the target hardware (the target's basis gates) and contain only two-qubit gates, according to the constraints specified by the connectivity of the hardware (the target's coupling map).  When using Qiskit Runtime to submit a job to an IBM Quantum backend, the circuits must adhere to the backend's ISA. 

Reviewer: ```suggestion
### Instruction set architecture
In addition to reducing the depth and complexity of quantum circuits, the transpiler is designed to transform the instructions contained in a given `QuantumCircuit` to obey the **Instruction Set Architecture** (**ISA**) of a particular backend.  Circuits obeying the ISA consist only of gates that can be run on the target hardware (the target's basis gates) and contain only two-qubit gates, according to the constraints specified by the connectivity of the hardware (the target's coupling map).  When using Qiskit Runtime to submit a job to an IBM Quantum&trade; backend, the circuits must adhere to the backend's ISA. 
```

Reviewer: sentence case and trademark

> In addition to reducing the depth and complexity of quantum circuits, the transpiler is designed to transform the instructions contained in a given `QuantumCircuit` to obey the **Instruction Set Architecture** (**ISA**) of a particular backend.  Circuits obeying the ISA consist only of gates that can be run on the target hardware (the target's basis gates) and contain only two-qubit gates, according to the constraints specified by the connectivity of the hardware (the target's coupling map).  When using Qiskit Runtime to submit a job to an IBM Quantum backend, the circuits must adhere to the backend's ISA. 

Reviewer 2: > consist only of gates that can be run on the target hardware

The definition here is a little narrow if read as `(instructions in circuit)` \subset `(basis gates)`. Can we add something that makes it clear that ISA circuits may include other instructions supported by the backend? e.g. `measure`, `reset`, control-flow operations (`if`, `else`, `for`, etc.)

Reviewer 2: > contain only two-qubit gates, according to

the grammatical construction here also reads as if the circuit can only consist of two-qubit gates.

Author: Just made a few edits based on these suggestions.  Let me know what you think.

> In addition to reducing the depth and complexity of quantum circuits, the transpiler is designed to transform the instructions contained in a given `QuantumCircuit` to obey the **Instruction Set Architecture** (**ISA**) of a particular backend.  Circuits obeying the ISA will consist only of gates which can be executed on the target hardware (i.e. the target's basis gates) and will only contain two-qubit gates according to the constraints specified by the connectivity of the hardware (i.e. the target's coupling map).  Circuits which follow the ISA of a particular backend is also a requirement when submitting jobs to one of IBM Quantum's backends through Qiskit IBM Runtime.

Reviewer 3: ```suggestion
In addition to reducing the depth and complexity of quantum circuits, the transpiler is designed to transform the instructions contained in a given `QuantumCircuit` to obey the **Instruction Set Architecture** (**ISA**) of a particular backend.  Circuits obeying the ISA consist only of gates that can be run on the target hardware (the target's basis gates) and contain only two-qubit gates, according to the constraints specified by the connectivity of the hardware (the target's coupling map).  When using Qiskit Runtime to submit a job to an IBM Quantum system, the circuits must adhere to the system's ISA. 
```
